### DISCUSSION TITLE
Can I dryrun renovate (using docker) against a local (on disk) git repo?

### DISCUSSION THREAD

**Discussion by nielsbasjes** [renovatebot/renovate#20581] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I want to experiment with more advanced config (like regex managers) for renovate locally on my Linux machine.
For this purpose I would like to be able to run renovate (preferably from docker) in a dry-run setup against a local git repo, just to see if it comes up with the correct changes. 
I want to do this locally because then 
1) I get much faster feedback compared to running it in a real environment like Gitlab or Github.
2) I want to be able to quickly change my config and my project files.

### What you tried so far.

I had a good look at the documentation: https://docs.renovatebot.com/examples/self-hosting/ and https://docs.renovatebot.com/modules/platform/ .
So far it seems renovate can (not very surprising) only work with a hosted remote git hosting platform where it can then put the merge requests.

I have not been able to figure out how to run it (if possible from the provided docker image) against a local git repo.

**Comment by viceice** [comment]
not possible yet, you can use baseBranches and dryRun to test things.

so use base branches to configure a test branch and then run renovate in dryRun.

you can configure the base branch on local bot side.

https://docs.renovatebot.com/configuration-options/#usebasebranchconfig

**Comment by rarkins** [comment, marked as answer]
- #3609
